extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/MultiStageQueryContext.java

```diff
@@ -152,6 +154,7 @@ public class MultiStageQueryContext
   public static final String CTX_ARRAY_INGEST_MODE = "arrayIngestMode";
   public static final ArrayIngestMode DEFAULT_ARRAY_INGEST_MODE = ArrayIngestMode.MVD;
 
+  public static final String CTX_SKIP_TYPE_VERIFICATION = "skipTypeVerification";
```

Comment:
Would we have docs for this parameter? If so, it'll be another parameter that we'd have to support in MSQ. If not, then I don't see a way for users to set this parameter.

Reply:
Can we do without this parameter, since the user can use ARRAY_TO_MV or MV_TO_ARRAY functions to preserve the behaviour?

Reply:
I don't think `ARRAY_TO_MV` and `MV_TO_ARRAY` would be helpful, since the purpose of this parameter is to allow someone to explicitly decide to ingest mismatched types. Whereas, `ARRAY_TO_MV` and `MV_TO_ARRAY` are more something that users could use to make types line up properly.

I'm not sure why anyone would want to bypass this validation and ingest mismatched types, to be honest. But I added this since I felt weird about having a type validation with no way to bypass it. I do agree it's not really ideal for it to be a new context parameter. I just couldn't think of another approach to allowing validation bypass for specific columns, without a table-level catalog.

I wonder if you have any thoughts here? (or anyone else?)

Reply:
the context flag seems ok to me, especially since i don't think we're quite yet in a place where a lot of people will want to be using a mix of mvds and arrays, we are missing optimized versions of `MV_TO_ARRAY` and `ARRAY_TO_MV` to make things a lot more efficient to use a mix of stuff.

I feel like once we've transitioned to `array` being default `arrayIngestMode`, given some time for it to settle, and then eventually remove `arrayIngestMode` to drop all of the other behaviors, we might be able to consider removing this validation in favor of pushing it to the catalog since the old habits would be gone and at that point it should also be a lot less painful to work with a mix of stuff. But we can revisit once `arrayIngestMode` is removed.